submit/submit

```diff
@@ -0,0 +1,534 @@
+
+    # We got a successful HTTP response. It worked.
+    # But check that we indeed received a submission ID.
+
+    try:
+        submission = json.loads(response.text)
+    except json.decoder.JSONDecodeError as e:
+        error('parsing REST API output: {}'.format(e))
+
+    if (not isinstance(submission, dict)
+        or not 'id' in submission
+        or not isinstance(submission['id'], str)):
+        error('REST API returned unexpected JSON data')
+
+    print('Submission received, id = s{}'.format(submission['id']))
+
+version_text = '''
+submit -- part of DOMjudge
+Written by the DOMjudge developers
+
+DOMjudge comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
+are welcome to redistribute it under certain conditions.  See the GNU
+General Public Licence for details.
+'''
+
+loglevels = {
+    'DEBUG': logging.DEBUG,
+    'INFO': logging.INFO,
+    'WARNING': logging.WARNING,
+    'ERROR': logging.ERROR,
+    'CRITICAL': logging.CRITICAL,
+}
+
+# Note: we set add_help to false since we can only print the help text after parsing flags, since the help contains data needed from the API
+parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description='Submit a solution for a problem.', add_help=False)
+parser.add_argument('--version', action='version', version=version_text, help='output version information and exit')
+parser.add_argument('-h', '--help', help='display this help and exit', action='store_true')
+parser.add_argument('-c', '--contest', help='''submit for contest with ID or short name CONTEST.
+    Defaults to the value of the
+    environment variable 'SUBMITCONTEST'.
+    Mandatory when more than one contest is active.''')
+parser.add_argument('-p', '--problem', help='submit for problem with ID or label PROBLEM')
+parser.add_argument('-l', '--language', help='submit in language with ID LANGUAGE')
+parser.add_argument('-e', '--entry_point', help='set an explicit entry_point, e.g. the java main class')
+parser.add_argument('-v', '--verbose', help='increase verbosity', choices=loglevels.keys(), nargs='?', const='INFO', default='WARNING')
+parser.add_argument('-q', '--quiet', help='suppress warning/info messages', action='store_true')
+parser.add_argument('-y', '--assume-yes', help='suppress user input and assume yes', action='store_true')
+parser.add_argument('-u', '--url', help='''submit to server with base address URL
+    (should not be necessary for normal use)''')
+parser.add_argument('filename', nargs='*', help='filename(s) to submit')
+
+args = parser.parse_args()
+
+verbosity = args.verbose
+if args.quiet:
+    verbosity = 'ERROR'
+
+logging.basicConfig(format='[%(asctime)s] {}[{}]: %(message)s'.format(sys.argv[0], os.getpid()), datefmt='%b %d %H:%M:%S', level=loglevels[verbosity])
+
+contest_id = ''
+
+if 'SUBMITBASEURL' in os.environ:
+    baseurl = os.environ['SUBMITBASEURL']
+if 'SUBMITCONTEST' in os.environ:
+    contest_id = os.environ['SUBMITCONTEST']
```

Comment:
Undocmented feature?

Reply:
Bats uses it. Should we document it somewhere?

Reply:
I guess? I'm a fan of either having a feature and documenting it or not having it at all.